### ISSUE TITLE
[Bug]: --theme-editor-sync running infinite loop

### ISSUE THREAD

**Issue by bigskillet** [Shopify/cli#4107]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Running `shopify theme dev --store --theme-editor-sync` should sync once on start.

### Actual behavior

Sync gets stuck in an infinite loop: 
<img width="389" alt="Screenshot 2024-06-20 at 11 01 23 AM" src="https://github.com/Shopify/cli/assets/682013/af380c57-f638-4a63-8a48-d881a30e702b">

### Verbose output

```shell
Sorry, but --verbose runs in an infinite loop when combined with --theme-editor-sync. I'm not exactly sure what to paste here.
```

### Reproduction steps

1. `shopify theme dev --store --theme-editor-sync`

### Operating System

macOS Sonoma 14.5

### Shopify CLI version

3.62.0

**Comment by dreamer-cloud** [comment]
+1, same issue after updating to 3.62.0

**Comment by agentfitz** [comment]
+1, same

**Comment by stijns96** [comment]
Hi there 👋🏻 

It probably has to due with the new trailing comma support. There are 2 solutions at the moment:

- [Add a comma after the closing curly bracket](https://github.com/Shopify/cli/issues/3222#issuecomment-2179233154)
- [Empty the file that is stuck in an infinite loop](https://github.com/Shopify/cli/issues/3222#issuecomment-2117182606)

This is always happening with the settings_schema.json . If you have this with another file, you most likely have an illegal unicode in your code.

**Comment by JamiesonRoberts** [comment]
As a note, something really does need to be done about this, because most code editors these days do auto formatting of files, which does not support the trailing comma as its not part of the spec of JSON. This is causing a massive headache of having to manually copy files when making changes, rather than letting the CLI handle those changes

**Comment by karreiro** [comment]
👋 Hey everyone,

Thank you for your input and all the details about this issue. We've released a new version of the Shopify CLI some minutes ago with the fix (`3.63.2`). Please, if this problem persists, don't hesitate to reach out and share your `--verbose` output here.

Thanks again for reporting!

**Comment by lesterdefreitas** [comment]
<img width="695" alt="image" src="https://github.com/Shopify/cli/assets/12233287/72731c4a-957f-42ba-9367-0574099e6253">

Oh no... @karreiro

**Comment by lesterdefreitas** [comment]
Ah, awesome, thanks dude!

**Comment by DanielDanielsson** [comment]
Oh sweet baby jesus.. Thank you guys!

**Comment by dspeake** [comment]
Same here. We have some stores that are doing this across 100+ template files.
I personally don't think it was ever fully fixed, I still had issues with some `.context.json` files.

It feels to me as though it's to do with the new JSON comments that are added to the top of the file? Perhaps one side is trying to add them and the other side is trying to remove them? Just a thought.

**Comment by rcasimmons** [comment]
> Same here. We have some stores that are doing this across 100+ template files. I personally don't think it was ever fully fixed, I still had issues with some `.context.json` files.
> 
> It feels to me as though it's to do with the new JSON comments that are added to the top of the file? Perhaps one side is trying to add them and the other side is trying to remove them? Just a thought.

Yeah that could be a plausible solution.

The weird thing is that before 3.67.X everything was working as it should but you're right - I don't think it was fully addressed.

Hopefully @karreiro can shed some light :)

**Comment by stijns96** [comment]
I feel like the CLI is getting worse and worse, unfortunately. Almost every release has a bug that breaks the entire CLI. If not at startup, then a minute later. Hot reload is sometimes executed multiple times and debugging is a hopeless task...

I'm glad we finally don't have to use brew anymore, but the quality of the CLI really needs to go up. Also, developing for multistore is not yet fully optimised unfortunately

**Comment by ENG1N3X** [comment]
I had the same issue with version `3.67.2`

Adding a comma to all JSON files solved my problem

**Comment by agentfitz** [comment]
Yeah... what the heck? I just came back to my own codebase to make a small change and it is instantly going crazy, endlessly looping over json files. What gives?

Video evidence:
https://www.loom.com/share/71077c75b5d74c829a1bc3badda39fe7

**Comment by rcasimmons** [comment]
I can't get my --verbose log at the moment @karreiro but this still appears to be happening with the `--legacy` flag - I have to use due to an issue I've logged here:
https://github.com/Shopify/cli/issues/4548

**Comment by karreiro** [comment]
👋 Hey everyone,

It seems like a platform change has impacted the original fix applied to this issue.

If you're still facing the infinite loop in the most recent version of Shopify CLI, could you please share your `--verbose` output at https://github.com/Shopify/cli/issues/4537? With that, the team may get extra pointers about the issue you're facing. This issue should be addressed in Shopify CLI **3.67.3**, but we want to make sure that everyone is no longer facing it.

> I can't get my --verbose log at the moment @karreiro but this still appears to be happening with the --legacy flag

@rcasimmons, the `--legacy` flag was introduced as a convenient alternative to access the previous implementation of Shopify CLI commands (even with the most updated version of the CLI installed). However, the previous implementation of commands doesn't include recent fixes or features. So, removing the `--legacy` flag should resolve this issue. If it doesn't, please feel free to report that on https://github.com/Shopify/cli/issues/4537.

Thank you, everyone, for the quick feedback.
